src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -28,98 +25,114 @@ public XunitPerformanceHarness(string[] args)
 
         public void RunBenchmarks(string assemblyPath)
         {
             Validate(assemblyPath);
 
-            Action<string> collectOutputFilesCallback = fileName =>
-            {
+            Action<string> collectOutputFilesCallback = fileName => {
                 // FIXME: This will need safe guards when the client calls RunBenchmarks in different threads.
                 _outputFiles.Add(fileName);
             };
 
-            Action winRunner = () =>
-            {
+            Action winRunner = () => {
                 _runner(assemblyPath);
             };
-            Action nixRunner = () =>
-            {
+            Action nixRunner = () => {
                 _runner(assemblyPath);
-                ProcessResults(assemblyPath, Configuration.RunId, _outputDirectory, collectOutputFilesCallback);
+                ProcessResults(assemblyPath, Configuration.RunId, OutputDirectory, collectOutputFilesCallback);
             };
             Action runner = IsWindowsPlatform ? winRunner : nixRunner;
 
-            s_profiler(assemblyPath, Configuration.RunId, _outputDirectory, runner, collectOutputFilesCallback);
+            s_profiler(assemblyPath, Configuration.RunId, OutputDirectory, runner, collectOutputFilesCallback);
         }
 
-        public void RunScenario(ProcessStartInfo processStartInfo, Action<ScenarioTestConfiguration> preIterationDel,
-            Action<ScenarioTestConfiguration> postIterationDel, Func<ScenarioTestConfiguration, ScenarioBenchmark> teardownDel)
+        /// <summary>
+        /// Executes the benchmark scenario specified by the parameter
+        /// containing the process start information.<br/>
+        /// The process component will wait, for the benchmark scenario to exit,
+        /// the time specified on the configuration argument.<br/>
+        /// If the benchmark scenario has not exited, then it will immediately
+        /// stop the associated process, and a TimeoutException will be thrown.
+        /// </summary>
+        /// <param name="processStartInfo">The ProcessStartInfo that contains the information that is used to start the benchmark scenario process.</param>
+        /// <param name="preIterationDelegate">The action that will be executed before every benchmark scenario execution.</param>
+        /// <param name="postIterationDelegate">The action that will be executed after every benchmark scenario execution.</param>
+        /// <param name="teardownDelegate">The action that will be executed after running all benchmark scenario iterations.</param>
+        /// <param name="configuration">ScenarioConfiguration object that defined the scenario execution.</param>
+        public void RunScenario(
+            ProcessStartInfo processStartInfo,
+            Action preIterationDelegate,
+            Action postIterationDelegate,
+            Func<ScenarioBenchmark> teardownDelegate,
+            ScenarioConfiguration configuration)
```

Comment:
Do you want to have a default value for ScenarioConfiguration?  Such as ScenarioConfiguration.Default which can contain the defaults for all the config values.